molgenis-audit/src/main/java/org/molgenis/audit/AuditEventPublisher.java

```diff
@@ -23,14 +25,21 @@ public AuditEventPublisher(ApplicationEventPublisher applicationEventPublisher)
    *
    * @param principal the principal's name
    * @param type the audit event type
-   * @param data additional information pertaining to the event
+   * @param data a modifiable map with additional information pertaining to the event
```

Comment:
Feels like premature optimization to me.
Allow any Map and use e.g. ImmutableMap.builder() to create a copy with the transaction id in it